packages/client/lib/client/socket.ts

```diff
@@ -28,15 +28,15 @@ export interface RedisSocketCommonOptions {
   reconnectStrategy?: false | number | ((retries: number, cause: Error) => false | Error | number);
 }
 
-type RedisNetSocketOptions = Partial<net.SocketConnectOpts> & {
+export interface RedisNetConnectOpts extends Omit<Partial<net.TcpNetConnectOpts>, 'keepAlive'>, Partial<net.IpcNetConnectOpts>, RedisSocketCommonOptions {
```

Comment:
I'm pretty sure this solves #2679, need to check (and if it does, cherry-pick it)